### ISSUE TITLE
Rule for ordering reference links

### ISSUE THREAD

**Issue by nschonni** [nodejs/remark-preset-lint-node#140]
This is more style over function, but got the idea from @aduh95 in #139

Basic premise would be to flag:
```md
Lorem impsom [Link A][], [Link B][]

[Link B]: http://example.com/bar
[Link A]: http://example.com/foo
```

Would flag that the `Link A` reference is before the `Link B` reference in the bottom.

Reasoning is, some pages the reference section gets quite large, and an alphabetical list is easier to visually parse when looking for one to update.

**Comment by aduh95** [comment]
Something like this has been implemented in node core in https://github.com/nodejs/node/pull/35244. However, the `checkLinks.js` script doesn't check documentation files (but instead, all the other markdown files that are not cover `make lint-md`).

If https://github.com/nodejs/node/pull/35191 lands, that would change, enforcing de facto the ordering of references for the docs, so I'm not sure it's worth investing time making a lint rule on top of it.

**Comment by nschonni** [comment]
@aduh95 interesting, thanks! It would be interesting to find/write a remark-lint rule though, because then other repos that use this preset could take advantage as well

**Comment by Trott** [comment]
Is this fixed by #188 or are they different things?

**Comment by aduh95** [comment]
Yes, #188 do enforce ordering of reference links, and would typically flag the example in the OP as invalid. 
Closing, @nschonni please reopen if you meant something else.
